autoconfigure/storage-elasticsearch-aws/src/main/java/zipkin/autoconfigure/storage/elasticsearch/aws/ElasticsearchDomainEndpoint.java

```diff
@@ -13,73 +13,75 @@
 import java.io.IOException;
 import java.util.Collections;
 import java.util.List;
+import java.util.concurrent.CompletionException;
 import java.util.logging.Logger;
-import okhttp3.HttpUrl;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.Response;
-import okio.Buffer;
-import zipkin2.elasticsearch.ElasticsearchStorage;
+import zipkin2.elasticsearch.ElasticsearchStorage.HostsSupplier;
 
-import static zipkin2.elasticsearch.internal.JsonReaders.enterPath;
+import static zipkin.autoconfigure.storage.elasticsearch.aws.ZipkinElasticsearchAwsStorageAutoConfiguration.JSON;
 
-final class ElasticsearchDomainEndpoint implements ElasticsearchStorage.HostsSupplier {
+final class ElasticsearchDomainEndpoint implements HostsSupplier {
   static final Logger log = Logger.getLogger(ElasticsearchDomainEndpoint.class.getName());
 
-  final OkHttpClient client;
-  final Request describeElasticsearchDomain;
+  final HttpClient client;
+  final AggregatedHttpRequest describeElasticsearchDomain;
 
-  ElasticsearchDomainEndpoint(OkHttpClient client, HttpUrl baseUrl, String domain) {
+  ElasticsearchDomainEndpoint(HttpClient client, String domain) {
     if (client == null) throw new NullPointerException("client == null");
-    if (baseUrl == null) throw new NullPointerException("baseUrl == null");
     if (domain == null) throw new NullPointerException("domain == null");
     this.client = client;
     this.describeElasticsearchDomain =
-        new Request.Builder()
-            .url(baseUrl.newBuilder("2015-01-01/es/domain").addPathSegment(domain).build())
-            .tag("get-es-domain")
-            .build();
+        AggregatedHttpRequest.of(HttpMethod.GET, "/2015-01-01/es/domain/" + domain);
   }
 
-  @Override
-  public List<String> get() {
-    try (Response response = client.newCall(describeElasticsearchDomain).execute()) {
-      String body = response.body().string();
-      if (!response.isSuccessful()) {
-        String message =
-            describeElasticsearchDomain.url().encodedPath()
-                + " failed with status "
-                + response.code();
-        if (!body.isEmpty()) message += ": " + body;
-        throw new IllegalStateException(message);
-      }
-      JsonReader endpointReader = JsonReader.of(new Buffer().writeUtf8(body));
-      endpointReader = enterPath(endpointReader, "DomainStatus", "Endpoints");
-      if (endpointReader != null) endpointReader = enterPath(endpointReader, "vpc");
-
-      if (endpointReader == null) {
-        endpointReader =
-            enterPath(JsonReader.of(new Buffer().writeUtf8(body)), "DomainStatus", "Endpoint");
-      }
+  @Override public List<String> get() {
+    HttpStatus status;
+    String body;
+    try {
+      AggregatedHttpResponse res = client.execute(describeElasticsearchDomain).aggregate().join();
+      status = res.status();
+      // As the domain endpoint is read only once per startup. We don't worry about pooling etc.
```

Comment:
@anuraaga let me know if this is true or not

Reply:
It's true